script.js

```diff
@@ -2,11 +2,164 @@
+
+  let description = document.getElementsByClassName("episodeDescription")[index];
+  description.innerHTML = `${list[index].summary}`;
+}
+
+function getSeasonAndEpisodeNumber(list, index) {
+  let seasonNumber = `S${(list[index].season < 10 ? "0" : "") + list[index].season}`
+  let episodeNumber = `E${(list[index].number < 10 ? "0" : "") + list[index].number}`
+
+  return `${seasonNumber + episodeNumber}`
+}
+
+//level 200
+function displayResults() {
+  let pageHeader = document.querySelector("header")
+
+  let pageNavigation = document.createElement("nav");
+  pageHeader.appendChild(pageNavigation);
+
+  let searchBar = document.createElement("input");
+  searchBar.setAttribute("type", "text");
+  searchBar.setAttribute("placeholder", "Search through the episodes");
+  searchBar.setAttribute("id", "mySearchBar");
+  pageNavigation.appendChild(searchBar)
+
+  let searchResult = document.createElement("label")
+  searchResult.classList.add("mySearchResult")
+
+  let getEpisodesOnPage = document.getElementsByClassName("episodeWrapper");
+  searchResult.innerText = `Displaying ${getEpisodesOnPage.length} episodes `;
+  pageNavigation.appendChild(searchResult)
+
+  searchBar.addEventListener("keyup", searchThroughEpisodes)
+
+  function searchThroughEpisodes(e) {
+    let searchString = e.target.value.toUpperCase();
+    let allEpisodesOnPage = Array.from(getEpisodesOnPage);
+    let newEpisodes = [];
+
+    allEpisodesOnPage.forEach(episode => {
+
+      if (episode.innerText.toUpperCase().includes(searchString)) {
+        episode.style.display = "";
+        newEpisodes.push(episode);
+        searchResult.innerText = `Displaying ${newEpisodes.length}/${getEpisodesOnPage.length} episodes `;
+      } else {
+        episode.style.display = "none";
+        searchResult.innerText = `Displaying ${newEpisodes.length}/${getEpisodesOnPage.length} episodes `;
+      }
+      if (searchString.length === 0) {                                                                      //I have this condition, so when you type "winter" for example and get a result of 10/73, then when you delete it all you want the result to be 73 and not 73/73
+        searchResult.innerText = `Displaying ${getEpisodesOnPage.length} episodes `;
+      }
+    });
+
+  }
 }
 
+//level 300
+function selectEpisode() {
+  let chooseEpisode = document.createElement("label");
+  chooseEpisode.innerText = "Choose an episode";
+  chooseEpisode.classList.add("myChooseEpisode")                                                           //setAttribute can also be used to add class.
+
+  let selectAnEpisode = document.createElement("select");
+  selectAnEpisode.setAttribute("id", "episodes")
+
+  let pageNavigation = document.querySelector("nav")                                                      //I declared this variable again (there is one in level 200), did not want to have anything in the global scope, so not sure if it is best practice?
```

Comment:
Yes, you should try to avoid having variables in global scope. What you are doing here is the best practice 👍